src/Tribe/Status/Manager.php

```diff
@@ -292,10 +296,71 @@ public function get_status_options( $commerce ) {
 	public function get_providers_status_classes( $commerce ) {
 
 		if ( ! isset( $this->statuses[ $commerce ] ) ) {
-			return array();
+			return [];
 		}
 
 		return $this->statuses[ $commerce ];
 
 	}
+
+	/**
+	 * Get the Completed Status by Commerce Provider Class Name
+	 *
+	 * @since TBD
+	 *
+	 * @param string|object $provider_name an object or string of a commerce main class name
+	 *
+	 * @return array
+	 */
+	public function get_completed_status_by_provider_name( $provider_name ) {
+
+		if ( is_object( $provider_name ) ) {
+			$provider_name = get_class( $provider_name );
+		}
+
+		if ( ! isset( $this->module_slugs[ $provider_name ] ) ) {
+			return [];
+		}
+
+		$trigger_statuses = [];
+		$abbvetied_name   = $this->module_slugs[ $provider_name ];
+
+		$filtered_statuses = wp_list_filter( $this->statuses[ $abbvetied_name ]->statuses, [
+			'count_completed' => true,
+		] );
+
+
+		foreach ( $filtered_statuses as $status ) {
+			$trigger_statuses[] = $status->provider_name;
+
+			if ( ! empty( $status->additional_names ) ) {
+				$trigger_statuses = $this->add_additional_names_to_array( $trigger_statuses, $status->additional_names );
```

Comment:
What's the difference between this and array_merge()?

Reply:
From what I read it is best to avoid using array_merge in loops. Not sure if that would matter here, but it is habit.